doc/adrs/138_add_collapse_and_expand_services.adoc

```diff
@@ -0,0 +1,62 @@
+= ADR-138 Add a collapse and expand services
+
+== Context
+
+Sirius Web doesn't provide a way for specifiers to define tools that collapse/expand nodes as part of other operations (e.g. to expand a collapsed node when a child node is created).
+
+We will add a new `diagramServices` variable  with services to collapse/expand nodes from AQL expressions and Java services.
+
+== Decision
+
+We will create a new interface `IDiagramServices` in `org.eclipse.sirius.components.collaborative.diagrams.api` with the following signature:
+```java
+public interface IDiagramServices {
+    void collapse(IDiagramServicesContext context, List<Node> nodes);
+    void expand(IDiagramServicesContext context, List<Node> nodes);
+}
+```
+
+We will add a new variable `diagramServices : IDiagramServiceContext` in the `VariableManager`.
```

Comment:
We have tons of things name `XxxContext` just use something like diagram service. I'm trying to limit the usage of diagram context which we had to provide as an API a bit too quickly in my mind so I'm not fan of another context. This term should be necessary for our consumers.

Reply:
renamed to IDiagramService